### PR TITLE
chore: refactor sql runner redux state to have entire sql chart data

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: 

### Description:

We need to store the entire saved SQL chart in the redux state as we need most (if not all) information for different parts of the page. The `savedSqlChart` property should be treated as "read only".

The properties that are editable(name, SQL, config) should also be stored as separate properties in the redux state as treated as the "unsaved/dirty" state.

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/features/sqlRunner/store/sqlRunnerSlice.ts [hunk 1]
```diff
@@ -24,23 +24,13 @@ export const DEFAULT_NAME = 'Untitled SQL Query';
...
     activeTable: string | undefined;
-    savedSqlUuid: string | undefined;
-    slug: string | undefined;
-    space:
-        | {
-              uuid: string;
-              name: string;
-          }
-        | undefined;
+    savedSqlChart: SqlChart | undefined;
```
Comment (joaoviana) [inline packages/frontend/src/features/sqlRunner/store/sqlRunnerSlice.ts:27]: can we then do this?
```suggestion
    readonly savedSqlChart: SqlChart | undefined;
```

**Review by joaoviana — APPROVED** [review]
One suggestion!

File: packages/frontend/src/features/sqlRunner/store/sqlRunnerSlice.ts (hunk 1 above)
Comment (joaoviana) [reply packages/frontend/src/features/sqlRunner/store/sqlRunnerSlice.ts:27]: Ignore ^ it has to be set anyway

**Review by magnew — APPROVED** [review]
👍

**Pull request merged**